Clarification about mailserver trust requirement

staheri14 opened the pull request:
Addresses https://github.com/vacp2p/rfc/issues/436
- This PR adds the privacy and trust consideration regarding mailserver clients (the security considerations about other capabilities are already provided, however, the mailserver part is missing).
- It also includes some clarification about the disclosure of a light node's IP/ID to its direct connections.

staheri14 committed:
clarification about mailservers trust assumption

staheri14 committed:
some revisions

staheri14 committed:
ID/IP

staheri14 committed:
minor fix

oskarth reviewed (commented):
LGTM; @jakubgs?

jakubgs reviewed (approved):
This sounds good, and I see nothing wrong with it, but I was hoping we would also update this doc:
https://github.com/vacp2p/rfc/blob/dc97d001b78c8f132570fcb80455b1fd20f0d0f7/content/docs/rfcs/8/README.md#security-considerations

staheri14 committed:
updates mail server security considerations

staheri14 committed:
Merge remote-tracking branch 'origin/master' into 6-mailserver-privacy-concern

staheri14 commented:
> This sounds good, and I see nothing wrong with it, but I was hoping we would also update this doc:
> https://github.com/vacp2p/rfc/blob/dc97d001b78c8f132570fcb80455b1fd20f0d0f7/content/docs/rfcs/8/README.md#security-considerations

I have added an explainer to the mailserver specs in https://github.com/vacp2p/rfc/pull/450/commits/a8fd785024cb14d2b1e00847b67c09905f32ad2e
Let me know if that resolves the ambiguity @jakubgs 

@oskarth I have added an explanation to the mailserver specs about the similarity between the security issues in the mailserver-client case and light nodes-direct peers (see 
https://github.com/vacp2p/rfc/pull/450/commits/a8fd785024cb14d2b1e00847b67c09905f32ad2e), however, I am not quite sure whether it is a good practice or not (it is hard to track such cross specs references and discussion in the future). Let me know your thoughts @oskarth.

oskarth reviewed (approved):
Thanks! I think in general it is better to have it described in one canonical place, then in other spec do a one sentence summary + link to other spec for more details. Since this spec isn't actively being developed, I think what we have now is fine.

jakubgs reviewed (approved):
Thank you, looks good.
